packages/coinstac-common/src/models/db-listener.js

```diff
@@ -42,8 +42,11 @@ class DBListener extends EventEmitter {
    */
   destroy() {
     this.removeAllListeners();
-    this.changes.removeAllListeners();
-    this.changes.cancel();
+
+    return new Promise(resolve => {
+      this.changes.on('complete', resolve);
```

Comment:
This is _totally_ undocumented. PouchDB’s `Cancel` emitter fires a `cancel` event. Something _somewhere_ catches this and emits a `complete` event when all requests have finished.